exporter/awsemfexporter/config.go

```diff
@@ -64,6 +64,12 @@ type Config struct {
 	// TODO: we can support directing output to a file (in the future) while customer specifies a file path here.
 	OutputDestination string `mapstructure:"output_destination"`
 
+
+	// CreateEKSFargateKubernetesObject is an option to reformat certin metric labels so that they take the form of a high level object
+	// The end result will make the labels look like those coming out of ECS and be more easily injected into cloudwatch
```

Comment:
I can't understand what does this new config do from the description. What does 'high level object' refer to?

Also, the config naming of  'CreateEKSFargateKubernetesObject' is also weird. It doesn't look like an attribute for Config.

Reply:
where does `ParseJSONEncodedAttributeValues` being used in your changes?

Reply:
ParseJSONEncodedAttributeValues is being used implicitly as the code I added to the exporter creates a new label "kubernetes" and sets the value to the JSON string of the created object. We had considered having it so that setting the CreateEKSFargateKubernetesObject flag would automatically add "kubernetes" to ParseJSONEncodedAttributeValues but decided it would go against the design philosophy.

The high level object is just the way I was referring the the nested kubernetes object because it will be a high-level key-value pair. The flag simply tells the exporter whether or not to add in the new object. As for the naming of the flag it was just what made the most sense to me - I'm happy to rename it if something else is preferred.

Reply:
Do we need an option for this? If cloudwatch handles kubernetes objects better with this format we may as well just always format right? Alternatively, have you considered improving the backend instead of adding more complexity to clients such as the collector? Perhaps it's a good first step on the way to native OTLP support in Cloudwatch

Reply:
@anuraaga @hossain-rayhan We decided against doing this by default in case some customers had backwards compatibility issues with having the extra key in the labels map. I'm not certain why the feature was implemented here rather than in CW or EKS, Rayhan might know more.

Reply:
Hi @anuraaga , I think we don't want to make this a default behavior because-
- This feature is very specific to EKS Fargate customers. For EKS EC2 customers, they will have the special metadata (`kubernetes`) with the metric itself and that would be done from the receiver. Also this exporter is being used by other customers like ECS and Batch (maybe) where we don't need this. 
- Having a flag will give the customers an option to opt-in which is good in my opinion. We may have some EKS Fargate customers who don't want Container Insights experience but want to ship metrics to CW. This flag will be helpful to support both use cases. 

For @mxiamxia's comment, can we rename the flag like, `eks_fargate_container_insights_enabled`? This name would be generic and we can apply this for other changes specific to EKS Fargate Container Insights (if needed).